packages/amplify_authenticator/lib/src/views/confirm_signup_viewmodel.dart

```diff
@@ -12,44 +14,52 @@ class ConfirmSignUpViewModel extends BaseViewModel {
 
   GlobalKey<FormState> get formKey => _formKey;
 
-  String? _username;
+  late String username;
 
   String? _code;
 
   void setUsername(String value) {
-    _username = value;
+    username = value;
   }
 
   void setCode(String value) {
     _code = value;
   }
 
   Future<void> confirm() async {
-    if (!_formKey.currentState!.validate()) {
-      return;
-    }
     setBusy(true);
     final confirmation =
-        AuthConfirmSignUpData(code: _code!.trim(), username: _username!.trim());
+        AuthConfirmSignUpData(code: _code!.trim(), username: username.trim());
 
     _authBloc.authEvent.add(AuthConfirmSignUp(confirmation));
 
     await Future.any([
-      _authBloc.exceptions.first,
+      _authBloc.exceptions!.first,
       _authBloc.stream.first,
     ]);
     setBusy(false);
   }
 
+//This method is called directly from this view due to the fact that
+//its result doesn't form part of the authenticator state machine.
+  Future<void> resendSignUpCode() async {
+    try {
+      await Amplify.Auth.resendSignUpCode(username: username);
+    } on AmplifyException catch (e) {
+      print(e);
+      _authBloc.exceptionsSink!.add(AuthenticatorException(e.message));
+    }
+  }
+
   // Screens
 
   void goToSignIn() {
     clean();
+    _authBloc.exceptionsSink!.add(null);
```

Comment:
Prefer to create a method in `AuthBloc` called `clearException` or something which does this.